File: 2025/Company/Company.Web/Modules/Company/PersonPhone/PersonPhoneColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.PersonPhone")]
[BasedOnRow(typeof(PersonPhoneRow), CheckNames = true)]
public class PersonPhoneColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public string BusinessEntityPersonType { get; set; }
    [EditLink]
    public string PhoneNumber { get; set; }
    public string PhoneNumberTypeName { get; set; }
    public short IsActive { get; set; }
    public int TenantId { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/PersonPhoneColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PersonPhoneRow } from "./PersonPhoneRow";

export interface PersonPhoneColumns {
    BusinessEntityPersonType: Column<PersonPhoneRow>;
    PhoneNumber: Column<PersonPhoneRow>;
    PhoneNumberTypeName: Column<PersonPhoneRow>;
    IsActive: Column<PersonPhoneRow>;
    TenantId: Column<PersonPhoneRow>;
}

export class PersonPhoneColumns extends ColumnsBase<PersonPhoneRow> {
    static readonly columnsKey = 'Company.PersonPhone';
    static readonly Fields = fieldsProxy<PersonPhoneColumns>();
}